app/models/conversation.rb

```diff
@@ -17,12 +17,93 @@ def timestamp
     end
 
     def response_object
+        arena = self.arena
         return {
             :id => id,
             :title => title,
+            :summaries => [
+                {
+                    :author => arena.user2.response_object,
+                    :text => self.summary_of_first
+                },
+                {
+                    :author => arena.user1.response_object,
+                    :text => self.summary_of_second
+                }
+            ],
+            :resolution => {
+                :text => self.resolution,
+                :state => self.resolution_state,
+                :updated_time => self.resolution_updated_at,
+                :updated_by_id => self.resolution_updated_by_id
+            },
             :posts => posts.map { |post|
                 post.response_object
-            }
+            },
         }
     end
+
+    def user_did_edit_summary(user_id, text)
+        arena = self.arena
+        if arena.is_first_user(user_id)
+            self.summary_of_second = text
+        elsif arena.is_second_user(user_id)
+            self.summary_of_first = text
+        else
+            return false
+        end
+        save
+        return true
+    end
+
+    def user_did_approve_summary(user_id)
+        arena = self.arena
+        if !summary_of_first.empty? && arena.is_first_user(user_id)
+            posts.create :author => arena.user2, :text => summary_of_first, :post_type => :summary
+            self.summary_of_first = ""
+        elsif !summary_of_second.empty? && arena.is_second_user(user_id)
+            posts.create :author => arena.user1, :text => summary_of_second, :post_type => :summary
+            self.summary_of_second = ""
+        else
+            return false
+        end
+        save
+        return true
+    end
+
+    def user_did_edit_resolution(user_id, text)
+        # TODO Check that user_id is one of the two users in the conversation.
+        # TODO Also, implement a way of storing edit history so users don't stomp each other.
+        if self.resolution_state == :consensus_reached
+            return false
+        end
+        self.resolution = text
+        self.resolution_state = :in_progress
+        self.resolution_updated_at = Time.now
+        self.resolution_updated_by_id = user_id
+        save
+        return true
+    end
+
+    def user_did_approve_resolution(user_id)
+        if allow_user_to_approve_consensus user_id
+            self.resolution_state = :consensus_reached
+            posts.create :author => arena.user1, :text => resolution, :post_type => :resolution
```

Comment:
The post is registered under user1, but that's not what's displayed. What's displayed to the user is:
You and ____:
x hours ago
Lorem ipsum.........